app/Http/ViewHelpers/Project/ProjectTasksViewHelper.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace App\Http\ViewHelpers\Project;
+
+use App\Helpers\DateHelper;
+use App\Models\Company\Company;
+use App\Models\Company\Project;
+use Illuminate\Support\Collection;
+use App\Models\Company\ProjectTask;
+use App\Models\Company\ProjectTaskList;
+
+class ProjectTasksViewHelper
+{
+    /**
+     * Collection containing the information about the tasks in the project.
+     * This collection contains all the tasks as well as all the tasks lists.
+     *
+     * @param Project $project
+     * @return array
+     */
+    public static function index(Project $project): array
+    {
+        $company = $project->company;
+        $tasks = $project->tasks()
+            ->with('list')
+            ->with('assignee')
+            ->with('author')
+            ->latest()
+            ->get();
+
+        // the goal of the following is to first display tasks without lists,
+        // and after this, tasks with lists, grouped by lists.
+        // the trick is to do this with a single query, as we don’t want to do
+        // multiple queries to slow down the loading speed of the page.
+
+        $tasksWithoutLists = $tasks->filter(function ($task) {
+            return is_null($task->project_task_list_id);
+        });
+        $tasksWithoutListsCollection = collect([]);
+        foreach ($tasksWithoutLists as $task) {
+            $tasksWithoutListsCollection->push(self::getTaskInfo($task, $company));
+        }
```

Comment:
You can combine the `foreach` with:
```suggestion
        $tasksWithoutListsCollection = $tasksWithoutLists = $tasks->filter(function ($task) {
            return is_null($task->project_task_list_id);
        })->map(function($task) use ($company) {
            return self::getTaskInfo($task, $company));
        })
```

Reply:
It’s almost ok without `$tasksWithoutLists =` in your suggestion.

Reply:
Or... no I’m wrong.